src/provider/rsa_keymgmt.c

```diff
@@ -1439,16 +1577,30 @@ static int ibmca_keymgmt_rsa_get_params(void *vkey, OSSL_PARAM params[])
     }
 
     /* Private key parts */
-    rc = ibmca_keymgmt_rsa_priv_as_bn(key, &p, &q, &dp, &dq, &qinv);
+    rc = ibmca_keymgmt_rsa_priv_crt_as_bn(key, &p, &q, &dp, &dq, &qinv);
     if (rc == 1) {
-        rc = ibmca_keymgmt_rsa_calc_priv_d(key, n, e, p, q, &d);
-        if (rc == 0)
-            goto out;
+        /* CRT format */
+        rc = ibmca_keymgmt_rsa_priv_me_as_bn(key, &d);
```

Comment:
Why do you first try to calculate d from the me sub-struct? Is it faster?

Reply:
I don't. 
I first try to get it from the substructure via ibmca_keymgmt_rsa_priv_me_as_bn() and only if that fails, I calculate it via ibmca_keymgmt_rsa_calc_priv_d().